src/internals.jl

```diff
@@ -2,6 +2,20 @@ module Internals
 
 import StableTasks: @spawn, @spawnat, @fetch, @fetchfrom, StableTask, AtomicRef
 
+if (
+    (@isdefined Core) &&
+    (Core isa Module) &&
+    isdefined(Core, :Compiler) &&
+    (Core.Compiler isa Module) &&
+    isdefined(Core.Compiler, :return_type) &&
+    applicable(Core.Compiler.return_type, identity, Tuple{})
+)
+    infer_return_type(@nospecialize f::Any) = Core.Compiler.return_type(f, Tuple{})
```

Comment:
one last question, is there much reason for the `@nospecialize` here? Won't that just potentially block inference?

Reply:
`@nospecialize` doesn't block inference, that's `@nospecializeinfer`. My reasoning is that the computation only happens at macro instantiation time, so there's not much reason to specialize.

Reply:
https://docs.julialang.org/en/v1/base/base/#Base.@nospecializeinfer